openquake/hazardlib/gsim/abrahamson_2018.py

```diff
@@ -0,0 +1,380 @@
+    2.000     8.00    400.0   -0.299  0.64875  -0.550   0.68  -0.00510  0.0070  -0.487  -0.0610  -0.301  -0.20847000000000  -0.08503000000000   0.61  0.450  0.4301  0.4310        -0.3          0.3      -0.30        0.30
+    2.500     7.95    400.0    0.000  0.08221  -0.501   0.68  -0.00465  0.0060  -0.770  -0.0711  -0.313  -0.20955750000000  -0.05003166666667   0.61  0.450  0.3795  0.3795        -0.3          0.3      -0.30        0.30
+    3.000     7.90    400.0    0.000 -0.36926  -0.460   0.68  -0.00430  0.0052  -0.700  -0.0798  -0.323  -0.21660250000000  -0.01508333333333   0.61  0.450  0.3280  0.3280        -0.3          0.3      -0.30        0.30
+    4.000     7.85    400.0    0.000 -1.03439  -0.455   0.68  -0.00390  0.0040  -0.607  -0.0935  -0.282  -0.06406000000000   0.06239166666667   0.61  0.450  0.2505  0.2550        -0.3          0.3      -0.30        0.30
+    5.000     7.80    400.0    0.000 -1.51967  -0.450   0.73  -0.00370  0.0030  -0.540  -0.0980  -0.250   0.06459750000000   0.06338666666667   0.61  0.450  0.2000  0.2000        -0.3          0.3      -0.30        0.30
+    6.000     7.80    400.0    0.000 -1.81025  -0.450   0.78  -0.00357  0.0022  -0.479  -0.0980  -0.250   0.09014000000000   0.09661000000000   0.61  0.450  0.2000  0.2000        -0.3          0.3      -0.30        0.30
+    7.500     7.80    400.0    0.000 -2.17269  -0.450   0.84  -0.00340  0.0013  -0.393  -0.0980  -0.250   0.13793250000000   0.12577500000000   0.61  0.450  0.2000  0.2000        -0.3          0.3      -0.30        0.30
+    10.00     7.80    400.0    0.000 -2.71182  -0.450   0.93  -0.00327  0.0000  -0.350  -0.0980  -0.250   0.31434000000000   0.23962833333333   0.61  0.450  0.2000  0.2000        -0.3          0.3      -0.30        0.30
+    """)
+
+    CONSTANTS = {"n": 1.18,
+                 "c": 1.88,
+                 "C4": 10.0,
+                 "a3": 0.10,
+                 "a5": 0.0,
+                 "a9": 0.4,
+                 "a10": 1.73,
+                 "C1slab": 7.2,
+                 "phiamp": 0.3}
+
+
+class AbrahamsonEtAl2018SInterHigh(AbrahamsonEtAl2018SInter):
+    """
+    Abrahamson et al (2018) subduction interface GMPE with the positive
+    epistemic adjustment factor applied
+    """
+    EPISTEMIC_ADJUSTMENT = "SINTER_HIGH"
+
+
+class AbrahamsonEtAl2018SInterLow(AbrahamsonEtAl2018SInter):
+    """
+    Abrahamson et al (2018) subduction interface GMPE with the negative
+    epistemic adjustment factor applied
+    """
+    EPISTEMIC_ADJUSTMENT = "SINTER_LOW"
+
+
+class AbrahamsonEtAl2018SSlab(AbrahamsonEtAl2018SInter):
+    """
+    Abrahamson et al. (2018) updated "BC Hydro" subduction GMPE for application
+    to subduction in-slab earthquakes.
+    """
+
+    #: Required rupture parameters for the in-slab model are magnitude and top
+    # of rupture depth
+    REQUIRES_RUPTURE_PARAMETERS = set(('mag', 'ztor'))
+
+    #: Cascadia adjustment factor
+    CASCADIA_ADJUSTMENT = "adj_slab"
```

Comment:
Hi there Graeme
Is this missing 
    `DEFINED_FOR_TECTONIC_REGION_TYPE = const.TRT.SUBDUCTION_INTRASLAB`
?
Chris
Or similar

Reply:
Yes, sorry I overlooked that. Now fixed. Cheers